### PR TITLE
feat(systemd): setting systemd service with Restart=always property by default

### PR DESCRIPTION
**Pull request by Azraeht**

### CODE REVIEW

File: vars/main.yml [hunk 1]
```diff
@@ -2,3 +2,4 @@
 es_package_url: "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch"
 reserved_xpack_users: ["elastic","kibana","logstash_system"]
 sysd_config_file: "/etc/systemd/system/elasticsearch.service.d/override.conf"
+sysd_always_restart: True
```
Comment (alsyia) [inline vars/main.yml]: I think this should be a `default` and not a `var` if we want it to be overridable (maybe there are cases in which we don't want ES to autorestart? I'm not sure). The wording you used in the README makes it sound like you want it to be configurable. ^^

File: templates/override.conf.j2 [hunk 2]
```diff
@@ -4,4 +4,8 @@ LimitMEMLOCK = infinity
 {% endif %}
 {% if es_max_open_files is defined %}
 LimitNOFILE = {{ es_max_open_files }}
-{% endif %}
\ No newline at end of file
+{% endif %}
+{% if sysd_always_restart %}
+Restart=always
```
Comment (alsyia) [inline templates/override.conf.j2]: FYI systemd recommends using `Restart=on-failure` to do what we want. See [here](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=). But that won't make any difference in the current situation :)

File: vars/main.yml (hunk 1 above)
Comment (Azraeht) [reply vars/main.yml]: You're rigth! Fixed!

File: templates/override.conf.j2 (hunk 2 above)
Comment (Azraeht) [reply templates/override.conf.j2]: `on-failure` is more adapted right. Fixed.

**Pull request merged**